src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java

```diff
@@ -853,7 +853,7 @@ public final void markDirty()
      */
     public void registerBlockPosition(@NotNull Block block, @NotNull final BlockPos pos)
     {
-        if (block instanceof BlockContainer)
+        if (block instanceof BlockContainer || block instanceof BlockMinecoloniesRack)
```

Comment:
If the block is already placed here, we should check the TE, it has a method hasCapabilty, which allows to check for IItemHandlers, regardless of block or mod.

This would in my humble opinion be the better solution (if we have the TE), if we donnot have the TE this is a good hotfix.